### ISSUE TITLE
salt cloud map_run

### ISSUE THREAD

**Issue by sandeepbalouria** [saltstack/salt#18973]
Hi ,
I am using salt cloud API to deploy a map. I have been manually editing cloud/**init** and adding 
opt['hard]=False in def _opts_defaults to make it work but i just want to see if there is a bug or am i missing something.

without  any modification salt 2014.7.0  map-run fails with 
 if self.opts['hard']:
KeyError: 'hard'

If i pass keyword argument to map_run like sclient.map_run(cluster_map,hard= False)

if self.opts['hard']:
            if self.opts['enable_hard_maps'] is False:
                raise SaltCloudSystemExit(
                    'The --hard map can be extremely dangerous to use, '
                    'and therefore must explicitly be enabled in the main '
                    'configuration file, by setting \'enable_hard_maps\' '
                    'to True'
                )

```
        # Hard maps are enabled, Look for the items to delete.
        ret['destroy'] = exist.difference(defined)
    return ret
```

I get the above exception. I don't want hard maps . How can i do this .
Thanks in advance.

Regards,
Sandeep

**Comment by cachedout** [comment]
Re-ping @techhat

**Comment by eradman** [comment]
This was fixed in commit 1599e1625cd

**Comment by cachedout** [comment]
@eradman Thanks for following-up. Much appreciated. I will close this.
